src/common/hooks/callbacks/use-decode-request-callback.ts

```diff
@@ -0,0 +1,20 @@
+import { useRecoilCallback } from 'recoil';
+import { requestTokenStore } from '@store/recoil/transaction';
+import { useLocation } from 'react-router-dom';
+
+export function useDecodeRequestCallback() {
+  const location = useLocation();
+  return useRecoilCallback(
+    ({ set }) => () => {
+      const urlParams = new URLSearchParams(location.search);
+      const requestToken = urlParams.get('request');
```

Comment:
Does any validation need to happen here on the input? Could this be a malicious payload?

Reply:
More is done here: #1092